.github/workflows/ci.yaml

```diff
@@ -0,0 +1,63 @@
+name: CI
+on:
+  pull_request: {}
+  push:
+    branches:
+      - main
+
+jobs:
+  go-build:
+    # This job effectively exists to ensure that the code can still be built
+    # with the proposed changes.
+    name: Go - Build
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: read
+      checks: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - run: make build
+  go-unit-test:
+    name: Go - Unit Tests
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: read
+      checks: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Unit Tests
+        run: go test -race -shuffle=on -coverprofile=coverage.txt ./...
+  go-vet:
```

Comment:
Usually, I prefer to use `golangci-lint`, since this includes a wider set of linters. It also has a fairly nice support for GitHub Actions annotation mechanism. I held off adding it in this PR since I think it would be good to discuss this with other maintainers before adding it.

Reply:
Yep, makes sense.

I'm okay with `golangci-lint` too. We can discuss with other maintainers as well.

Let's merge this, run it for a few times; and then we can amend as neeeded.